### PR TITLE
wallet: cleanup cached amount and input mine check code

### PR DESCRIPTION
**Pull request by furszy**
Another wallet's code garbage collector work. Part of the `mapWallet` encapsulation goal.

Focused on the following points:

1) Remove always true `fUseCache` argument from `CachedTxGetImmatureCredit`, `CachedTxGetImmatureWatchOnly` and `CachedTxGetAvailableCredit`.
2) Remove always false `recalculate` argument from `GetCachableAmount`.
3) Merge `CachedTxGetImmatureCredit` and `CachedTxGetImmatureWatchOnlyCredit` as they do share the exact same code.
4) Clean `InputIsMine` method; use `GetWalletTx` instead of access the wallet's map directly.
5) Clean `AllInputsMine` method; use `InputIsMine` instead of duplicate the exact same code internally.

### CODE REVIEW

File: src/wallet/test/wallet_tests.cpp [hunk 1]
```diff
@@ -350,13 +350,13 @@ BOOST_FIXTURE_TEST_CASE(coin_mark_dirty_immature_credit, TestChain100Setup)
...
     // cache the current immature credit amount, which is 0.
-    BOOST_CHECK_EQUAL(CachedTxGetImmatureCredit(wallet, wtx), 0);
+    BOOST_CHECK_EQUAL(CachedTxGetImmatureCredit(wallet, wtx, ISMINE_SPENDABLE), 0);
 
     // Invalidate the cached value, add the key, and make sure a new immature
     // credit amount is calculated.
     wtx.MarkDirty();
     AddKey(wallet, coinbaseKey);
-    BOOST_CHECK_EQUAL(CachedTxGetImmatureCredit(wallet, wtx), 50*COIN);
+    BOOST_CHECK_EQUAL(CachedTxGetImmatureCredit(wallet, wtx, ISMINE_SPENDABLE), 50*COIN);
```
Comment (aureleoules) [inline src/wallet/test/wallet_tests.cpp:359]: nit
```suggestion
    BOOST_CHECK_EQUAL(CachedTxGetImmatureCredit(wallet, wtx, ISMINE_SPENDABLE), 50 * COIN);
```

**Review by aureleoules — COMMENTED** [review]
ACK ba2fa8302dc3816d5b9584907135f91789846d65.
Looks better!
I checked that the refactor does not change the behavior of the code.

**Review by theStack — APPROVED** [review]
Code-review ACK ba2fa8302dc3816d5b9584907135f91789846d65

Nice refactoring!

**Comment by furszy** [comment]
per feedback, have inlined `AllInputsMine` return statement.

**Review by theStack — APPROVED** [review]
re-ACK 47ea70fbb85fefeb4de9d3142a11596d292eab9b

**Comment by furszy** [comment]
hey @jonatack @aureleoules, friendly ping here.

**Comment by aureleoules** [comment]
re-ACK 47ea70fbb85fefeb4de9d3142a11596d292eab9b

**Comment by achow101** [comment]
ACK 47ea70fbb85fefeb4de9d3142a11596d292eab9b

**Pull request merged**